handler/handler.go

```diff
@@ -1,15 +1,24 @@
 package handler
 
 import (
+	"github.com/gin-gonic/gin"
 	"github.com/murilogilfelpeto/ps-tag-onboarding-go/configuration"
 	"github.com/murilogilfelpeto/ps-tag-onboarding-go/service"
 )
 
-var (
-	logger *configuration.Logger
-)
+var logger = configuration.NewLogger("handler")
+
+type Handler interface {
+	Save(context *gin.Context)
+	FindById(context *gin.Context)
+}
+
+type handler struct {
+	service service.Service
+}
 
-func InitializeHandler() {
-	logger = configuration.GetLogger("handler")
-	service.Initialize()
+func NewHandler(service service.Service) Handler {
+	return &handler{
+		service: service,
+	}
```

Comment:
This is returning a user handler, why not call it `NewUserHandler`?

Reply:
This is returning a handler, but is being used just by userHandler

Reply:
After removing the interfaces in the code, It will return a UserHandler instead of a Handler